app/src/main/java/com/android/sample/model/map/LocationViewModel.kt

```diff
@@ -24,6 +25,10 @@ constructor(
   private val _currentLocation = MutableStateFlow<Location?>(null)
   val currentLocation: StateFlow<Location?> = _currentLocation
 
+  fun setCurrentLocation(location: Location?) {
```

Comment:
I will check for the way you manage the null cases, because you seem to willingly allow it here... Careful with this !!